samples/react-find-parker/src/webparts/findParker/components/FindParker/FindParker.tsx

```diff
@@ -0,0 +1,146 @@
+            <div className={styles.column}>
+              <p className={styles.title}>{strings.GameTitle}</p>
+              <p className={styles.label}>{strings.GameDescription}</p>
+            </div>
+          </div>
+          {!gameStarted ?
+            <div className={styles.row}>
+              <div className={styles.column}>
+                <DefaultButton onClick={() => this.startGame()}>{strings.StartGameButton}</DefaultButton>
+              </div>
+            </div>
+            :
+            <div className={styles.row}>
+              {!gameFinsihed ?
+                <div className={styles.column}>
+                  <p className={styles.label}>{strings.GameProgressELementsLabel} <strong>{numberOfElements}</strong></p>
+                  <p className={styles.label}>{strings.GameProgressLabel} <strong>{numberOfFoundElements}</strong></p>
+                </div>
+                :
+                <div className={styles.column}>
+                  <p className={styles.label}>{strings.EndGameMessage}</p>
+                  <DefaultButton onClick={() => this.restartGame()}>{strings.RestartGameButton}</DefaultButton>
+                </div>
+              }
+              <div className={styles.parkersContener}>
+                {elements}
+              </div>
+            </div>
+          }
+        </div>
+      </div>
+    );
+  }
+
+  private renderParkers(listOfParkers: Array<IParker>): void {
+    let items = [];
+    listOfParkers.forEach(parker => {
+      const style = {
+        top: parker.top,
+        left: parker.left,
+        transform: parker.transform
+      };
+      items.push(<img alt='pnpParker' src={Icon.svg} width='65' className={styles.parker} style={style} onClick={() => this.parkerFound(parker.id)} />);
+    });
+    this.setState({ elements: items });
+  }
+
+  private parkerFound(id: number): void {
+    let numberOfFoundElements = this.state.numberOfFoundElements + 1;
+    this.setState({ numberOfFoundElements: numberOfFoundElements });
+    let parkers = this.state.listOfParkers;
+    parkers = parkers.filter(parker => parker.id !== id);
+    this.setState({ listOfParkers: parkers });
+    this.renderParkers(parkers);
+
+    if (numberOfFoundElements === this.props.numberOfElements) {
+      this.setState({ gameFinsihed: true });
+    }
+  }
+
+  private startGame(): void {
+    this.setState({ gameStarted: true });
+    this.renderParkers(this.state.listOfParkers);
+  }
+
+  private restartGame(): void {
+    const { numberOfElements } = this.props;
+    this.setState({ gameFinsihed: false });
+    const listOfParkers = this.createListOfParkers(numberOfElements);
+    this.setState({
+      listOfParkers: listOfParkers,
+      numberOfFoundElements: 0
+    });
+    this.renderParkers(listOfParkers);
+  }
+
+  private createListOfParkers(numberOfElements: number): Array<IParker> {
+    const pageHeight: number = document.querySelector(Constants.mainPageContent).scrollHeight;
```

Comment:
You may want to check that the placeholder is available. In the workbench, these next two lines will generate errors and the rest of this function will fail.

Reply:
ok I added a double check in `componentDidMount` for this div and if it is not present I show a message to the user that something is not ok and he or she may not start the game.
this is how it looks/behaves now on workbench
![image](https://user-images.githubusercontent.com/58668583/140416647-155ec547-ce43-4a08-a294-5d0bc3a1ba3a.png)

and this is on a SharePoint Site
![image](https://user-images.githubusercontent.com/58668583/140416694-8ec49640-66aa-46d4-874e-815c592a7f5a.png)